tests/test_actions.cpp

```diff
@@ -254,6 +254,7 @@ TEST_F(ActionsTest, SetFromConst) {
   Data value(0xaba);
   SetField primitive;
   testActionFn.push_back_primitive(&primitive);
+  testActionFn.push_back_source_info(nullptr);
```

Comment:
if you make the source info an optional parameter to `push_back_primitive` as suggested above, you will not need to modify all these test files